composer/utils/module_surgery.py

```diff
@@ -101,13 +101,11 @@ def replace_module_classes(
         >>> replace_module_classes(module, policies, recurse_on_replacements=True)
         {Conv2d(1, 32, kernel_size=(3, 3), stride=(1, 1)): Linear(in_features=16, out_features=32, bias=True), Linear(in_features=16, out_features=32, bias=True): Linear(in_features=32, out_features=64, bias=True)}
 
-    .. warning::
-
-        When a module is replaced, any tensor values within the module are not copied over
-        to the new module even when the shape is identical. For example, if model weights
-        are initialized prior to calling this function, the initialized weights will not
-        be preserved in any replacements.
+    .. note::
```

Comment:
Let's add a `copy_weights: bool = False` flag to control this behavior. Some of our algorithms, such as BlurPool, handle the weight copy logic internally (e.g. see `from_Conv2d`) that we should not override automagically here.

Reply:
Agree that this should be a parameter, though I don't have an opinion on the default value.